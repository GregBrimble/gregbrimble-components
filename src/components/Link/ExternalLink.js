import React from "react"
import PropTypes from "prop-types"

class ExternalLink extends React.Component {
  render() {
    return (
      <a href={this.props.href} rel={`noopener noreferrer`} target={this.props.target}>
        {this.props.children}
      </a>
    )
  }
}

ExternalLink.propTypes = {
  href: PropTypes.string,
  target: PropTypes.string,
  children: PropTypes.any,
}

ExternalLink.defaultProps = {
  target: `_blank`,
}

export default ExternalLink
