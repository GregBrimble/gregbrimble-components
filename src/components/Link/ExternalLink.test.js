import React from "react"
import ExternalLink from "./ExternalLink"
import renderer from "react-test-renderer"

describe("ExternalLink", () => {
  it("renders properly", () => {
    const tree = renderer.create(<ExternalLink href={`https://google.com/`}>Link Text</ExternalLink>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
