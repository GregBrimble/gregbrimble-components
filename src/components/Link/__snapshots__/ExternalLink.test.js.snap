// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExternalLink renders properly 1`] = `
<a
  href="https://google.com/"
  rel="noopener noreferrer"
  target="_blank"
>
  Link Text
</a>
`;
